### ISSUE TITLE
Update Readme with information about the NuGet packages

### ISSUE THREAD

**Issue by rprouse** [nunit/nunit-console#93]
Move the NuGet package image/link into a table with a list of the NuGet packages and their current descriptions. We should also provide a link to the docs and possibly to the framework repository. We might also want to make it a bit clearer exactly what this repository contains, although that might be clear enough from the title 😄

**Comment by sergiorykov** [comment]
Please, provide information how can I use that banch of packages separately. The idea behind that to use extensions is clear but nuget it seems isn't supposed to be used as apt-get or chocolatey - to install console runner and extensions in a single location with availability in path or at least to combine them together in `tools` folder of main nupkg.

When I install NUnit.Console through command line it only downloads nupkg file without extracting all dependencies to `tools` folder as it was before separation. 
When I install NUnit.Console in csproj with unit tests all dependent packages are being installed separately
NUnit.Console.3.5.0\<empty tools only nupkg is here>
NUnit.ConsoleRunner.3.5.0\tools\nunit3-console.exe
NUnit.Extension.TeamCityEventListener.1.0.2\tools\teamcity-event-listener.dll

But I have separate packages.config with tools packages (such as NUnit.Console/MSBuildTasks/SpecFlow) only and I install them with `nuget install -ExcludeVersion` to use in build scripts. 

I can't find clean solution (aka super-duper-happy-path) with new NUnit.Console package.

Had to write script to join binaries
```
.nuget\nuget.exe install Tools\packages.config -OutputDirectory Tools -ExcludeVersion || exit 1

mkdir Tools\NUnit.Console\tools
xcopy Tools\NUnit.ConsoleRunner\tools\*.* Tools\NUnit.Console\tools /Y /V /I /F
xcopy Tools\NUnit.Extension.NUnitProjectLoader\tools\*.* Tools\NUnit.Console\tools /Y /V /I /F
xcopy Tools\NUnit.Extension.NUnitV2Driver\tools\*.* Tools\NUnit.Console\tools /Y /V /I /F
xcopy Tools\NUnit.Extension.NUnitV2ResultWriter\tools\*.* Tools\NUnit.Console\tools /Y /V /I /F
```

**Comment by ChrisMaddock** [comment]
If you run the console runner from `NUnit.ConsoleRunner.3.5.0\tools\nunit3-console.exe`, it should be able to locate the extensions in the other packages automatically. Is that not the case?

In `NUnit.ConsoleRunner\tools` you should have `nunit.nuget.addins`. That file should contain a list of relative paths which point to each of the extension dll's, in a default nuget set-up.

From the script you've posted, it looks like that should work - are they the full locations of your files, or is the structure more complicated than written? If the console can't locate the extensions, can you check if the path in `nunit.nuget.addins` are looking in the right place?

**Comment by sergiorykov** [comment]
Nice trick! 

It's not clear that it will work. I've migrated from NUnit.Console 3.0.x where all tools were located in a single package.
When I changed version to 3.5.0 `tools` vanished. I was really dissapointed - first thought was - package is corrupted :).

It will be used from build automation scripts mostly. That is why it is important to provide clear information where I will find my stuff.

If you'll add `tools\readme.txt` (in  NUnit.Console) with a migration plan like: change path to `NUnit.ConsoleRunner\tools` and add required additional extensions (link to nuget?q=NUnit.Extension.*) and they will be automagically found - with publishing in `Readme.md` of current repository. It will simplify usage and integration.

It's a new user experience even for me - I'm pretty close with nuget for several years - I've never seen such approach to addins for tools. 

Please, provide Super-Duper-Happy-Path of using it in scripts (as NancyFx proposes).

And thank you for your attention! 👍

**Comment by CharliePoole** [comment]
Chris is correct. But if you have trouble, consider simply using NUnit.Console as designed rather than copying files elsewhere. It takes a degree of understanding of the internals to move files around and still have everything work. Alternatively, use the msi or zip to create a central installation.

**Comment by sergiorykov** [comment]
Installing only Nunit.Console from command line nuget.exe will not work. I have to specify all dependent packages directly. It can't do it. 
They all will be added only when you add it in VS to project's packages. But when you add it from VS package manager for project they will be installed with version suffix - it's not useful for writing scripts. I prefer to setup path once.

**Comment by sergiorykov** [comment]
I agree - using ugly copying scripts isn't a solution. 

I've cleaned up dirty copying script and just changed path in build script:
```
<NUnitConsole>$(ToolsDir)\NUnit.Console**Runner**\tools\nunit3-console.exe</NUnitConsole>
```
Everything works as expected! 

but `packages.config` must contain all dependencies (for independent usage)

```
    <package id="NUnit.Console" version="3.5.0" />
	<package id="NUnit.ConsoleRunner" version="3.5.0" />
	<package id="NUnit.Extension.NUnitProjectLoader" version="3.5.0" />
	<package id="NUnit.Extension.NUnitV2Driver" version="3.5.0" />
	<package id="NUnit.Extension.NUnitV2ResultWriter" version="3.5.0" />
	<package id="NUnit.Extension.TeamCityEventListener" version="1.0.2" />
	<package id="NUnit.Extension.VSProjectLoader" version="3.5.0" /> 
```
`NUnit.Console` is useful only (it's only my point view of my personal experience :) - you've done great job!) to install from VS for unit tests csproj.
It's simply limitation of `nuget.exe` (it will download dependent packages) and dirty path to `packages\NUnit.ConsoleRunner.3.5.0\tools` directory I should add to build script for VS package registration in unit tests project.

**Comment by CharliePoole** [comment]
OK... can you summarize this in an issue for nunit/nunit-console? We're definitely getting beyond @rprouse''s issue about readme content!

Give an example of the nuget command-line you use, what you would like to see happen versus what actually happens. This will be helpful because we are very aware that the detection of extensions is quite fragile, depending as it does on how nuget names the files.

As an alternative to a new issue, you might just add a comment to nunit/nunit-console#4

**Comment by CharliePoole** [comment]
It has been so many releases, I forget how it was in 3.0 :-) I'll have to go back and check.

**Comment by sergiorykov** [comment]
Oups. Sorry. ^)

I wrote suggestions in https://github.com/nunit/nunit-console/issues/4 
and one question about extensions discovery...

**Comment by ChrisMaddock** [comment]
Done in https://github.com/nunit/nunit-console/pull/327
